js/src/forum/components/IndexPage.js

```diff
@@ -36,7 +36,7 @@ export default class IndexPage extends Page {
       app.cache.discussionList = null;
     }
 
-    const params = this.params();
+    const params = app.search.params();
```

Comment:
Just for clarification: In other (lower-level) components, we wouldn't want to access `app` within the component, right? And instead have these dependencies passed in as props?

Here it's okay because page components don't get any props passed in, and we're at the top level anyway?

Reply:
Yes. But the problem isn't that they are accessing app: that in of itself is fine. We want it to be extensible. No one's gonna instantiate an instance of Index Page somewhere random, but the lower level components might be reused by extensions in unexpected ways.

Reply:
> the problem isn't that they are accessing app: that in of itself is fine.

Well, it would break the unidirectional data flow we're envisioning, right?

Reply:
Not quite: My main goal is that the rendered content of the app can be considered the output of a somewhat pure, deterministic function, the input of which is the current state.

This means that if a component wants to change another component, it would do so by changing the state, and not by changing the component directly.

The only element I'm unsure of (and one we're gonna need to figure out) is transitional animations / direct DOM manipulations via JQuery, but we need to take this one step at a time.